src/backend/strategies.rs

```diff
@@ -661,6 +661,51 @@ pub async fn run_strategy_task<'s>(
                 let mut transition_count = 0;
                 let mut success_count = 0;
 
+                // Create asset lock proofs for all the identity creates and top ups
+                let num_identity_inserts = (strategy.identities_inserts.frequency.times_per_block_range.start) as u64;
+                let num_start_identities = strategy.start_identities.number_of_identities as u64;
+                let mut num_top_ups: u64 = 0;
+                for operation in &strategy.operations {
+                    if operation.op_type == OperationType::IdentityTopUp {
+                        num_top_ups += (operation.frequency.times_per_block_range.start) as u64;
+                    }
+                }
+                let num_asset_lock_proofs_needed = num_identity_inserts * num_blocks + num_start_identities + num_top_ups;
+                let mut asset_lock_proofs: Vec<(AssetLockProof, PrivateKey)> = Vec::new();
+                let mut wallet_lock = app_state.loaded_wallet.lock().await;
+                let num_available_utxos = match wallet_lock.clone().expect("No wallet loaded while getting asset lock proofs") {
+                    Wallet::SingleKeyWallet(SingleKeyWallet { utxos, .. }) => utxos.len(),
+                };
+                if num_available_utxos < num_asset_lock_proofs_needed.try_into().expect("Couldn't convert num_asset_lock_proofs_needed into usize") {
+                    // get more asset lock proofs
+                }
+                tracing::info!("Obtaining {num_asset_lock_proofs_needed} asset lock proofs for the strategy...");
+                let asset_lock_proof_time = Instant::now();
+                let mut num_asset_lock_proofs_obtained = 0;
+                for i in 0..(num_asset_lock_proofs_needed) {
+                    if let Some(wallet) = wallet_lock.as_mut() {
+                        match wallet.asset_lock_transaction(None, 2_000_000) {
+                            Ok((asset_lock_transaction, asset_lock_proof_private_key)) => {
+                                match AppState::broadcast_and_retrieve_asset_lock(sdk, &asset_lock_transaction, &wallet.receive_address()).await {
+                                    Ok(asset_lock_proof) => {
+                                        tracing::info!("Got {} asset lock proofs", i+1);
+                                        num_asset_lock_proofs_obtained += 1;
+                                        asset_lock_proofs.push((asset_lock_proof, asset_lock_proof_private_key));
+                                    },
+                                    Err(e) => tracing::error!("Error broadcasting and retrieving asset lock: {:?}", e),
+                                }
+                            },
+                            Err(e) => {
+                                tracing::error!("Error creating asset lock transaction: {:?}", e)
```

Comment:
Maybe we should break execution here? (return or maybe panic)?

Reply:
I'm not sure it's necessary here